### PR TITLE
Algolia updates

### PR DESCRIPTION
**Pull request by zephraph**
This update makes a series of changes requested by an algolia rep to enable docsearch. 

It includes
- making headings consistent
- ensuring headings use the right elements
- adding a sitemap
- add custom classes that algolia can hook into

Once we have this live they can start indexing us. Afterwards, the next big step will be to add the ui.

### CODE REVIEW

**Comment by damassi** [comment]
Nice - i'm amazed that they provide rep support for OSS.

**Comment by artsyit** [comment]
Deploy preview for *artsy-palette* ready!

Built with commit 5099b941ba0d71a2fee9574fa70c526877f2882b

https://deploy-preview-595--artsy-palette.netlify.com

File: packages/palette-docs/gatsby-config.js [hunk 1]
```diff
@@ -69,5 +70,6 @@ module.exports = {
     "gatsby-plugin-react-helmet",
     "gatsby-plugin-styled-components",
     "gatsby-plugin-typescript",
+    "gatsby-plugin-sitemap",
```
Comment (anandaroop) [inline packages/palette-docs/gatsby-config.js]: That was easy 😄

File: packages/palette-docs/src/components/GlobalComponents.jsx [hunk 2]
```diff
@@ -101,28 +101,40 @@ export const MarkdownComponents = {
   },
   h1: props => (
     <Box mb={5}>
-      <Serif size="8" color="black100">
+      <Serif element="h1" size="8" color="black100" className="DocSearch-lvl1">
```
Comment (anandaroop) [inline packages/palette-docs/src/components/GlobalComponents.jsx]: This seems retroactively like a good rationale for @dblandin's [suggestion](https://artsy.slack.com/archives/C9XJKPY9W/p1569448740010100)

File: packages/palette-docs/gatsby-config.js (hunk 1 above)
Comment (zephraph) [reply packages/palette-docs/gatsby-config.js]: Yeaaah, so good.

**Comment by artsyit** [comment]
:rocket: PR was released in v4.17.9 :rocket:

**Pull request merged**